cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -428,6 +426,9 @@ def _get_volume_stats(self):
                         # datastore
                         datastores[h.hubId] = vim_util.get_moref(h.hubId,
                                                                  "Datastore")
+                        datastore_profiles[datastores[h.hubId].value] = (
+                            profile['name']
+                        )
```

Comment:
That reads to me like you already foresee the case of a Datastore having multiple profiles. But five lines above, you skip over the datastore, if it is already associated though another storage profile.

Wouldn't it make sense to store for a Datastore all the matching datastore_profiles?

Reply:
no, I'm simply storing the association of the datastore to a profile name for later including it in the correct pool for stats.

Reply:
The association of datastore and profile is a many-to-many relationship. So, the code will only provide the "correct" pool, if the configuration parameters are chosen in a way, that the they are disjoint.

If I configure to profiles `vmware_storage_profile=profile-a,profile-b` and they both are applying to the same datastore then the results depend on the PBM to determine the order, assuming and python 3.5 or later, otherwise it is random, which of those show in the stats.

If we do not want to support the many-to-many relations, I would consider it sensible to at least warn about such a unsupported configuration.

Reply:
I'll update the code to store this in a list, since it's theoretically possible to have a datastore exist in more than 1 storage profile, but usually improbable.  The storage profile name is only used for an entry in the pool capabilities.   It's not really used anywhere.